src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs

```diff
@@ -179,5 +205,136 @@ private IActionResult RedirectToLocal(string returnUrl)
+        [AllowAnonymous]
+        public async Task<IActionResult> ForgotPassword()
+        {
+            if (!(await _siteService.GetSiteSettingsAsync()).As<RegistrationSettings>().EnableLostPassword)
+            {
+                return NotFound();
+            }
+
+            return View();
+        }
+
+        [HttpPost]
+        [AllowAnonymous]
+        public async Task<IActionResult> ForgotPassword(ForgotPasswordViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                var user = (User)await _userService.GetForgotPasswordUserAsync(model.UserIdentifier);
+                if (user != null)
+                {
+                    user.ResetToken = Convert.ToBase64String(Encoding.UTF8.GetBytes(user.ResetToken));
+                    // send email with callback link
+                    await SendResetTokenAsync(user);
+                }
+            }
+
+            // returns to confirmation page anyway: we don't want to let scrapers know if a username or an email exist
+            return RedirectToLocal(Url.Action("ForgotPasswordConfirmation"));
+        }
+        
+        [HttpGet]
+        [AllowAnonymous]
+        public IActionResult ForgotPasswordConfirmation()
+        {
+            return View();
+        }
+        
+        [HttpGet]
+        [AllowAnonymous]
+        public async Task<IActionResult> ResetPassword(string code = null)
+        {
+            if (!(await _siteService.GetSiteSettingsAsync()).As<RegistrationSettings>().EnableLostPassword)
+            {
+                return NotFound();
+            }
+            if (code == null)
+            {
+                //"A code must be supplied for password reset.";
+            }
+            return View(new ResetPasswordViewModel { ResetToken = code });
+        }
+
+        [HttpPost]
+        [AllowAnonymous]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> ResetPassword(ResetPasswordViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                if (await _userService.ResetPasswordAsync(model.Email, Encoding.UTF8.GetString(Convert.FromBase64String(model.ResetToken)), model.NewPassword, (key, message) => ModelState.AddModelError(key, message)))
+                {
+                    return RedirectToLocal(Url.Action("ResetPasswordConfirmation"));
+                }
+            }
+
+            return View(model);
+        }
+
+        [HttpGet]
+        [AllowAnonymous]
+        public IActionResult ResetPasswordConfirmation()
+        {
+            return View();
+        }
+
+        protected async Task<bool> SendResetTokenAsync(User user)
+        {
+            var message = new MailMessage();
+            message.IsBodyHtml = true;
+            message.To.Add(user.Email);
+            message.Subject = T["Reset your password"];
+
+            ISite site = await _siteService.GetSiteSettingsAsync();
+
+            string siteUrl = site.BaseUrl;
```

Comment:
Because the email is sent from a controller you don't need `BaseUrl`.
Create the link using `Url.Action` and the route values (not the string `"/OrchardCore.Users/Account/ResetPassword"`) then pass the `protocol: Request.Scheme` argument which will tell the helper to generate a fully qualified url using the current protocol.